source/server-discovery-and-monitoring/server-discovery-and-monitoring-tests.rst

```diff
@@ -178,3 +178,34 @@ Attempt a write to a collection.
 
 Outcome: Verify that the write succeeded or failed depending on existing
 driver behavior with respect to the starting topology.
+
+Monitors sleep at least minHeartbeatFreqencyMS between checks
```

Comment:
@ShaneHarvey This test is similar to the one you proposed, but it doesn't rely on setting heartbeatFrequencyMS to 500. This way it tests that minHeartbeatFrequencyMS is enforced even when immediate checks are being requested. Let me know what you think.

Reply:
This test seems reasonable to me. I'm testing it here: https://github.com/mongodb/mongo-python-driver/pull/524

A little disappointing that the test takes so long, could we reduce `times:10` to reduce the runtime?

Reply:
Reduced times to 5 and the expected time between 2 seconds and 3.5 seconds. I worry if we go too much shorter than that and we might lost some accuracy with the test.

As an aside, should we also add a test for the happy case to ensure heartbeatFrequencyMS is respected? Something like set it to 750ms and wait for 3.5 seconds and ensure at between 3 and 5 heartbeats occurred?

Reply:
Thanks, I'm testing the updated test here: https://evergreen.mongodb.com/version/5fc5a093d6d80a31f056c6c5

Hmmm, there may already be such a test for the happy path. I'll check for one. If there isn't one I'll open a new ticket for it instead of adding it here.

Reply:
I've double check are there isn't any existing test for the happy path behavior. I will create a drivers ticket for this tomorrow.